### PR TITLE
Implements ReadFiles for reading multiple files from the specified folder

### LINKED ISSUE #9: Implements ReadFiles for reading multiple files from the specified folder
# Summary

Would be nice o have a separate `ReadFiles` function to be able read files from the specified folder including subfolders and having match function for filtering by file type/extension  

## Basic example

``` go
type FileMatchFunc func(d fs.DirEntry) bool

func ReadFiles(root string, matchFun FileMatchFunc) ([]unstructured.Unstructured, error)
```

## Motivation

For tests it would be better to store multiple objects in separate files (for large objects it's a preferred way) and load them in one command without specify a separate files

### PR DESCRIPTION
**Pull request by dmvolod**
Adds ReadFiles function to be able read files from the specified folder including subfolders and having match function for filtering by file type/extension/whatever

fixes #9

### CODE REVIEW

**Comment by dmvolod** [comment]
//cc @afritzler

File: unstructuredutils/unstructuredutils.go [hunk 1]
```diff
@@ -45,6 +47,32 @@ func ReadFile(filename string) ([]unstructured.Unstructured, error) {
 	return Read(f)
 }
 
+// FileMatchFunc must match the filename against any specified name based criteria and return true if it matches, otherwise false
+type FileMatchFunc func(d fs.DirEntry) bool
+
+// ReadFiles reads unstructured objects from a folder with the given name (including sub folders)
+// and file name matched within FileMatchFunc.
+func ReadFiles(root string, matchFun FileMatchFunc) ([]unstructured.Unstructured, error) {
```
Comment (afritzler) [inline unstructuredutils/unstructuredutils.go:55]:
```suggestion
func ReadFiles(root string, matchFunc FileMatchFunc) ([]unstructured.Unstructured, error) {
```

**Review by afritzler — CHANGES_REQUESTED** [review]
Looks good to me! However, how about using https://pkg.go.dev/path/filepath#Glob instead of `WalkDir` for determining the objects location. So something like this:

```
files, err := filepath.Glob(srcDir + "/[a-Z0-9]")
```

**Comment by dmvolod** [comment]
> Looks good to me! However, how about using https://pkg.go.dev/path/filepath#Glob instead of `WalkDir` for determining the objects location. So something like this:

Good idea, @afritzler 
Implemented with the `filepath.Glob` and without separate filter function

**Review by afritzler — APPROVED** [review]

**Pull request merged**